src/cmd-buildextend-gcp

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python3
+# pylint: disable=C0103
+"""
+An operation that mutates a build by uploading to GCP,
+extending the meta.json with GCP image name.
+"""
+# NOTE: PYTHONUNBUFFERED is set in cmdlib.sh for unbuffered output
+
+import argparse
+import json
+import os
+import shutil
+import sys
+
+cosa_dir = os.path.dirname(os.path.abspath(__file__))
+sys.path.insert(0, cosa_dir)
+# pylint: disable=C0413
+from cmdlib import (
+        run_verbose,
+        sha256sum_file,
+        write_json)
+
+# Parse args and dispatch
+parser = argparse.ArgumentParser()
+parser.add_argument("--build", help="Build ID", required=True)
+parser.add_argument("--bucket", help="Storage account to write image to",
+                    default=os.environ.get("GCP_BUCKET", None))
+parser.add_argument("--gce", help="Use GCE as the platform ID instead of GCP",
+                    action="store_true",
+                    default=bool(os.environ.get("GCP_GCE_PLATFORM_ID", False)))
+parser.add_argument("--force", help="Replace existing images and upload",
+                    action="store_true",
+                    default=os.environ.get("GCP_FORCE", False))
+parser.add_argument("--json-key", help="GCP Service Account JSON Auth",
+                    default=os.environ.get("GCP_JSON_AUTH", None))
+parser.add_argument("--name-suffix", help="Append suffix to name",
+                    required=False)
+parser.add_argument("--project", help="GCP Project name",
+                    default=os.environ.get("GCP_PROJECT_NAME", None))
+args = parser.parse_args()
+
+# Argument checks for environment strings that are required
+arg_exp_str = "parameter '--{}' or envVar '{}' must be defined"
+if args.bucket is None:
+    raise Exception(arg_exp_str.format("bucket", "GCP_BUCKET"))
+if args.json_key is None:
+    raise Exception(arg_exp_str.format("json-key", "GCP_JSON_AUTH"))
+if args.project is None:
+    raise Exception(arg_exp_str.format("project", "GCP_PROJECT"))
+
+# Identify the builds
+builddir = os.path.join('builds', args.build)
+buildmeta_path = os.path.join(builddir, 'meta.json')
+with open(buildmeta_path) as f:
+    buildmeta = json.load(f)
+
+if 'gcp' in buildmeta['images'] and not args.force:
+  print("GCP image already built; use --force to rebuild")
+  sys.exit(1)
```

Comment:
Hmm, let's `exit 0` here? Being idempotent is useful in automated contexts (e.g. retrying a Jenkins job or something). Also matches `buildextend-installer`.